core/chains/evm/logpoller/log_poller.go

```diff
@@ -439,3 +440,34 @@ func (lp *LogPoller) LatestLogByEventSigWithConfs(eventSig common.Hash, address
 func (lp *LogPoller) LatestLogEventSigsAddrs(fromBlock int64, eventSigs []common.Hash, addresses []common.Address, qopts ...pg.QOpt) ([]Log, error) {
 	return lp.orm.LatestLogEventSigsAddrs(fromBlock, addresses, eventSigs, qopts...)
 }
+
+// IndexedLogs finds all the logs that have a topic value in topicValues at index topicIndex.
+func (lp *LogPoller) IndexedLogs(eventSig common.Hash, address common.Address, topicIndex int, topicValues []common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error) {
+	return lp.orm.SelectIndexedLogs(address, eventSig[:], topicIndex, topicValues, confs, qopts...)
+}
+
+// Index is 0 based.
+func (lp *LogPoller) LogsDataWordGreaterThan(eventSig common.Hash, address common.Address, wordIndex int, wordValueMin common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error) {
+	return lp.orm.SelectDataWordGreaterThan(address, eventSig[:], wordIndex, wordValueMin, confs, qopts...)
+}
+
+// Index is 0 based.
+func (lp *LogPoller) LogsDataWordRange(eventSig common.Hash, address common.Address, wordIndex int, wordValueMin, wordValueMax common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error) {
+	return lp.orm.SelectDataWordRange(address, eventSig[:], wordIndex, wordValueMin, wordValueMax, confs, qopts...)
+}
+
+// IndexedLogs finds all the logs that have a topic value greater than topicValueMin at index topicIndex.
+// Only works for integer topics.
+func (lp *LogPoller) IndexedLogsTopicGreaterThan(eventSig common.Hash, address common.Address, topicIndex int, topicValueMin common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error) {
+	return lp.orm.SelectIndexLogsTopicGreaterThan(address, eventSig[:], topicIndex, topicValueMin, confs, qopts...)
+}
+
+func (lp *LogPoller) IndexedLogsTopicRange(eventSig common.Hash, address common.Address, topicIndex int, topicValueMin common.Hash, topicValueMax common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error) {
+	return lp.orm.SelectIndexLogsTopicRange(address, eventSig[:], topicIndex, topicValueMin, topicValueMax, confs, qopts...)
+}
+
+func EvmWord(i uint64) common.Hash {
+	var b = make([]byte, 8)
+	binary.BigEndian.PutUint64(b, i)
+	return common.BigToHash(big.NewInt(0).SetBytes(b))
```

Comment:
Could call `BytesToHash` directly, no?
```suggestion
	return common.BytesToHash(b)
```